src/composables/__tests__/useMessageInfo.spec.js

```diff
@@ -0,0 +1,228 @@
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+import { computed } from 'vue'
+
+import { ATTENDEE, CONVERSATION, PARTICIPANT } from '../../constants.js'
+import { useConversationInfo } from '../useConversationInfo.js'
+import { useMessageInfo } from '../useMessageInfo.js'
+import { useStore } from '../useStore.js'
+
+jest.mock('@nextcloud/capabilities', () => ({
+	getCapabilities: jest.fn(() => ({
+		spreed: {
+			features: ['edit-messages'],
+		},
+	}))
+}))
+jest.mock('../useStore.js')
+jest.mock('../useConversationInfo.js')
+
+describe('message actions', () => {
+	let messageProps
+	let conversationProps
+	let mockConversationInfo
+	const TOKEN = 'XXTOKENXX'
+
+	beforeEach(() => {
+		messageProps = {
+			message: 'test message',
+			actorType: ATTENDEE.ACTOR_TYPE.USERS,
+			actorId: 'user-id-1',
+			actorDisplayName: 'user-display-name-1',
+			messageParameters: {},
+			id: 123,
+			isTemporary: false,
+			isFirstMessage: true,
+			isReplyable: true,
+			isTranslationAvailable: false,
+			canReact: true,
+			isReactionsMenuOpen: false,
+			isActionMenuOpen: false,
+			isEmojiPickerOpen: false,
+			isLastRead: false,
+			isForwarderOpen: false,
+			timestamp: new Date('2024-05-01 16:15:00').getTime() / 1000,
+			token: TOKEN,
+			systemMessage: '',
+			messageType: 'comment',
+			previousMessageId: 100,
+			participant: {
+				actorId: 'user-id-1',
+				actorType: ATTENDEE.ACTOR_TYPE.USERS,
+				participantType: PARTICIPANT.TYPE.USER,
+			},
+			showCommonReadIcon: true,
+			showSentIcon: true,
+			commonReadIconTooltip: '',
+			sentIconTooltip: '',
+		}
+		conversationProps = {
+			token: TOKEN,
+			lastCommonReadMessage: 0,
+			type: CONVERSATION.TYPE.GROUP,
+			readOnly: CONVERSATION.STATE.READ_WRITE,
+		}
+		useStore.mockReturnValue({
+			getters: {
+				message: () => messageProps,
+				conversation: () => conversationProps,
+				getActorId: () => 'user-id-1',
+				getActorType: () => ATTENDEE.ACTOR_TYPE.USERS,
+				isModerator: false,
+			},
+		})
+
+		mockConversationInfo = {
+			isOneToOneConversation: computed(() => false),
+			isConversationReadOnly: computed(() => false),
+			isConversationModifiable: computed(() => true),
+		}
+
+		useConversationInfo.mockReturnValue(mockConversationInfo)
+		jest.spyOn(global.Date, 'now')
+			.mockImplementation(() => '2024-05-01 17:00:00')
+	})
+
+	test('message is not deleteable when it is older than 6 hours and unlimited capability is disabled', () => {
+		// Arrange
+		messageProps.timestamp = new Date('2024-05-01 7:20:00').getTime() / 1000
+		// Act
+		const result = useMessageInfo(messageProps.token, messageProps.id)
+		// Assert
+		expect(result.isDeleteable.value).toBe(false)
+	})
+
+	test('message is not deleteable when the conversation is read-only', () => {
+		// Arrange
+		mockConversationInfo = {
+			isOneToOneConversation: computed(() => false),
+			isConversationReadOnly: computed(() => true),
+			isConversationModifiable: computed(() => true),
```

Comment:
You might want to add tests for `isConversationModifiable: computed(() => false)`